zk/stages/stage_batches_processor.go

```diff
@@ -307,11 +308,13 @@ func (p *BatchesProcessor) processFullBlock(blockEntry *types.FullL2Block) (endL
 		endLoop = true
 	}
 
-	// if we're above StepAfter, and we're at a step, move the stages on
-	if p.debugStep > 0 && p.debugStepAfter > 0 && blockEntry.L2BlockNumber > p.debugStepAfter {
-		if blockEntry.L2BlockNumber%p.debugStep == 0 {
-			log.Info(fmt.Sprintf("[%s] Debug step reached, stopping stage\n", p.logPrefix))
-			endLoop = true
+	if !p.debugStepExcludeBatches {
```

Comment:
We should add the else here which does:

- if progress >= limit { return nil }

This should allow the datastream stage to exit quickly without trying to reconnect for the head block!

Reply:
Isn't it done at line 306 above?

Reply:
My bad!